azurerm/resource_arm_container_group_test.go

```diff
@@ -0,0 +1,173 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMContainerGroup_basicLinux(t *testing.T) {
+	ri := acctest.RandInt()
+
+	name := fmt.Sprintf("acctestcontainergroup-%d", ri)
+	resourceGroupName := fmt.Sprintf("acctestRG-%d", ri)
+
+	config := testAccAzureRMContainerGroupBasicLinux(name, resourceGroupName, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMContainerGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMContainerGroupExists("azurerm_container_group.test"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMContainerGroup_basicWindows(t *testing.T) {
+	ri := acctest.RandInt()
+
+	name := fmt.Sprintf("acctestcontainergroup-%d", ri)
+	resourceGroupName := fmt.Sprintf("acctestRG-%d", ri)
+
+	config := testAccAzureRMContainerGroupBasicWindows(name, resourceGroupName, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMContainerGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMContainerGroupExists("azurerm_container_group.test"),
+				),
+			},
+		},
+	})
+}
+
+func testAccAzureRMContainerGroupBasicLinux(name string, resourceGroupName string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "%s"
+  location = "%s"
+}
+
+resource "azurerm_container_group" "test" {
+	
+	name = "%s"
+	location = "%s"
+	resource_group_name = "${azurerm_resource_group.test.name}"
+	ip_address_type="public"
+	os_type = "linux"
+  
+	container {
+	  name = "%s"
+	  image = "nginx:latest"
+	  cpu ="1"
+	  memory = "1.0"
+	  port = "80"
+#	  protocol = "TCP"
```

Comment:
is this commented out as it's not returned?

Reply:
Yes, if it's set, it causes a non-empty apply everytime. I think this is a bug in the API currently.